classes/plugininfo.php

```diff
@@ -92,17 +103,81 @@ public static function get_plugin_configuration_for_context(context $context, ar
 
         if ($courseid == 0) {
             $permissions = false;
+            $courseid = 0;
+            self::$itemdata = [];
+            self::$tradedata = [];
         } else {
             $manager = \block_stash\manager::get($courseid);
             $permissions = $manager->can_manage($USER->id);
             $courseid = $manager->get_courseid();
+
+            // Shoutout to crmpicco for isset guidance.
+            if (!isset(self::$itemdata)) {
+                self::$itemdata = array_map(
+                    fn(stdClass $item): array => (array)$item + [
+                        'imageurl' => moodle_url::make_pluginfile_url(
+                            context_course::instance($courseid)->id,
+                            'block_stash',
+                            'item',
+                            $item->id,
+                            '/',
+                            'image'
+                        )->out(false)
+                    ],
+                    $manager->get_drops_for_items_and_trade()['items']
```

Comment:
This function only returns data for drop data. This really requires all items and so I would recommend using manager->get_items();